nvtabular/framework_utils/tensorflow/feature_column_utils.py

```diff
@@ -0,0 +1,251 @@
+    for column in feature_columns:
+        parents = _get_parents(column)
+        base_columns.extend([col for col in parents if col not in base_columns])
+
+    cat_names, cont_names = [], []
+    for column in base_columns:
+        if isinstance(column, str):
+            # cross column input
+            # TODO: this means we only accept categorical inputs to
+            # cross? How do we generalize this? Probably speaks to
+            # the inefficiencies of feature columns as a schema
+            # representation
+            cat_names.extend(column)
+        elif isinstance(column, fc.CategoricalColumn):
+            cat_names.extend(column.key)
+        else:
+            cont_names.extend(column.key)
+    workflow = nvt.Workflow(cat_names=cat_names, cont_names=cont_names, label_name=[label_name])
+
+    _CATEGORIFY_COLUMNS = (fc.VocabularyListCategoricalColumn, fc.VocabularyFileCategoricalColumn)
+    categorifies, hashes, crosses, buckets, replaced_buckets = {}, {}, {}, {}, {}
+
+    numeric_columns = []
+    new_feature_columns = []
+    for column in feature_columns:
+        # TODO: check for shared embedding or weighted embedding columns?
+        # Do they just inherit from EmbeddingColumn?
+        if not isinstance(column, (fc.EmbeddingColumn, fc.IndicatorColumn)):
+            if isinstance(column, (fc.BucketizedColumn)):
+                # bucketized column being fed directly to model means it's
+                # implicitly wrapped into an indicator column
+                cat_column = column
+                embedding_dim = None
+            else:
+                # can this be anything else? I don't think so
+                assert isinstance(column, fc.NumericColumn)
+
+                # check to see if we've seen a bucketized column
+                # that gets fed by this feature. If we have, note
+                # that it shouldn't be replaced
+                if column.key in replaced_buckets:
+                    buckets[column.key] = replaced_buckets.pop(column.key)
+
+                numeric_columns.append(column)
+                continue
+        else:
+            cat_column = column.categorical_column
+
+            # use this to keep track of what should be embedding
+            # and what should be indicator, makes the bucketized
+            # checking easier
+            if isinstance(column, fc.EmbeddingColumn):
+                embedding_dim = column.dimension
+            else:
+                embedding_dim = None
+
+        if isinstance(cat_column, fc.BucketizedColumn):
+            key = cat_column.source_column.key
+
+            # check if the source numeric column is being fed
+            # directly to the model. Keep track of both the
+            # boundaries and embedding dim so that we can wrap
+            # with either indicator or embedding later
+            if key in [col.key for col in numeric_columns]:
+                buckets[key] = (column.boundaries, embedding_dim)
```

Comment:
I believe here and two lines below it should be `cat_column.boundaries`. It fails to find the `boundaries` column in my attempts to use this utility, and it passes if I replace `column.boundaries` with `cat_column.boundaries`.